template/combine.h

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2015 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef BRICKS_TEMPLATE_COMBINE_H
+#define BRICKS_TEMPLATE_COMBINE_H
+
+#include <tuple>
+#include <utility>
+
+namespace bricks {
+namespace metaprogramming {
+
+// `combine<std::tuple<A, B, C>>` == a `struct` that internally contains all `A`, `B` and `C`
+// via "has-a" inheritance, and exposes `operator()`, calls to which compile-time dispatched
+// to the instances of `A`, `B` or `C` respectively, based on type signature.
+// No matching invocation signature will result in compile-time error, same as the case
+// of more than one invocation signature matching the call.
+
+template <typename T>
+struct dispatch {
+  T has_a_instance;
```

Comment:
Fixed, in https://github.com/KnowSheet/Bricks/pull/160, https://github.com/dkorolev/Bricks/commit/40b020197aa9b8091443e1a933b70120eabba0cd.